components/remote_settings/src/context.rs

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use serde_json::{json, Map, Value};
+
+/// Remote settings context object
+///
+/// This is used to filter the records returned. We always fetch all `records` from the
+/// remote-settings storage. Some records could have a `filter_expression`.  If this is passed in
+/// and the record has a `filter_expression`, then only returns where the expression is true will
+/// be returned.
+///
+/// See https://remote-settings.readthedocs.io/en/latest/target-filters.html for details.
+#[derive(Debug, Clone, Default, uniffi::Record)]
+pub struct RemoteSettingsContext {
+    /// The delivery channel of the application (e.g "nightly")
+    #[uniffi(default = None)]
+    pub channel: Option<String>,
+    /// User visible version string (e.g. "1.0.3")
+    #[uniffi(default = None)]
+    pub app_version: Option<String>,
+    /// The locale of the application during initialization (e.g. "es-ES")
+    #[uniffi(default = None)]
+    pub locale: Option<String>,
+    /// The name of the operating system (e.g. "Android", "iOS", "Darwin", "Windows")
+    #[uniffi(default = None)]
+    pub os: Option<String>,
+    /// Form-factor of the device ("phone", "tablet", or "desktop")
+    #[uniffi(default = None)]
+    pub form_factor: Option<String>,
+    /// Country of the user
```

Comment:
How is this supposed to be populated by clients? 
I think that in quicksuggest uses something from Gecko that relies on a geojson file (`Region.sys.mjs`)
https://searchfox.org/mozilla-central/rev/88e49196740d9448f44506c2239a475c6c9eadc7/browser/components/urlbar/private/SuggestBackendRust.sys.mjs#444-447

Reply:
It's not totally clear in my mind, but I added some pointers based on what I know.